### PR TITLE
Week2

### PR DESCRIPTION
**Pull request by hanialkudsi**

### CODE REVIEW

**Comment by fedefu** [comment]
Thanks for the feedback @gajduk. Can you please implemet the feedback @hanialkudsi ?

File: Week2/language_test.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+create table test if not exists(country_name varchar(50),language varchar(50))
+delimiter //
+    CREATE TRIGGER languages_trigger BEFORE INSERT ON test FOR EACH ROW 
+    BEGIN  
+        DECLARE alert VARCHAR(200);  
+        DECLARE language_num INT; 
+        SET language_num = (select count(language) from test where country_name = new.country_name);
```
[the remark this answers is not in the document]
Comment (hanialkudsi) [reply Week2/language_test.sql]: thanks a lot man!

File: Week2/srever.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  if (err) throw err;
+  console.log('database connected');
+});
+
+// get the capital of a country
+// http://localhost:5055/capital/of/?country=spain
+app.get('/capital/of/?', (req, res) => {
+  const country = req.query.country;
+  var part1_1 = `select Name from city
+  where ID in ( SELECT capital FROM country WHERE name = '${country}' )`;
```
[the remark this answers is not in the document]
Comment (hanialkudsi) [reply Week2/srever.js]: i did change my code hopefully it's not a problem any more

File: Week2/srever.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    if (error) throw error;
+    res.send(resolve);
+  });
+});
+
+// get the languages in the region from : (use PostMan)
+// example : http://localhost:5055/languages/of/?region=south america
+app.get('/languages/of/?', (req, res) => {
+  const region = req.query.region;
+  var part1_2 = `select language from countrylanguage where CountryCode in (select code from country where region ='${region}')`;
```
[the remark this answers is not in the document]
Comment (hanialkudsi) [reply Week2/srever.js]: thanks man !
if there any more things need to be updated just let me know

File: Week2/srever.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  connection.query(part1_4, function(error, resolve) {
+    if (error) throw error;
+    res.send(resolve);
+  });
+});
+
+// get numbers of languages in a all continents
+// http://localhost:5055/continent/languages
+app.get('/continent/languages', (req, res) => {
+  var part1_5 = `select DISTINCT(continent),count(language) from country as co
```
[the remark this answers is not in the document]
Comment (hanialkudsi) [reply Week2/srever.js]: ohh ,it take me a long time to understand it but finally i did

**Comment by hanialkudsi** [comment]
changes have been done

**Comment by hanialkudsi** [comment]
i push the changes on week 3 homework by mistake
sorry guys
is it ok like this?

**Comment by gajduk** [comment]
I found the changes. No problem. Good job.

**Pull request closed without merging**